compiler/grainformat/reformat.re

```diff
@@ -1490,586 +2193,692 @@ and print_attributes = attributes =>
+                },
+              ])
+            | Some(guard) =>
+              let branch_guard_comments =
+                Comment_utils.get_comments_inside_location(
+                  ~location=guard.pexp_loc,
+                  comments,
+                );
+              let guard_doc =
+                Doc.group(
+                  print_expression(
+                    ~original_source,
+                    ~comments=branch_guard_comments,
+                    guard,
+                  ),
+                );
+              Doc.concat([
+                Doc.space,
+                Doc.text("when"),
+                Doc.space,
+                Doc.group(
+                  Doc.concat([
+                    Doc.ifBreaks(Doc.lparen, Doc.nil),
+                    Doc.indent(Doc.concat([Doc.softLine, guard_doc])),
+                    Doc.softLine,
+                    Doc.ifBreaks(Doc.rparen, Doc.nil),
+                  ]),
                 ),
-              ),
-              // keeping this as I think we can start to not add commas again
-              //  Doc.text(","),
-              //Doc.ifBreaks(Doc.text(","), Doc.nil),
-            ]),
-          ),
+                Doc.space,
+                Doc.text("=>"),
+                switch (branch.pmb_body.pexp_desc) {
+                | PExpBlock(_)
+                | PExpIf(_) =>
+                  Doc.concat([
+                    Doc.space,
+                    Doc.group(
+                      print_expression(
+                        ~original_source,
+                        ~comments=branch_comments,
+                        branch.pmb_body,
+                      ),
+                    ),
+                  ])
+
+                | _ =>
+                  Doc.indent(
+                    Doc.concat([
+                      Doc.line,
+                      Doc.group(
+                        print_expression(
+                          ~original_source,
+                          ~comments=branch_comments,
+                          branch.pmb_body,
+                        ),
+                      ),
+                    ]),
+                  )
+                },
+              ]);
+            },
+          ]),
+        );
+      };
+
+      let after_brace_comments =
+        Comment_utils.get_after_brace_comments(expr.pexp_loc, comments, None);
+      let cleaned_comments =
+        remove_used_comments(~all_comments=comments, after_brace_comments);
+
+      let items =
+        item_iterator(
+          ~previous=None,
+          ~get_loc,
+          ~print_item,
+          ~comments=cleaned_comments,
+          ~separator=Doc.comma,
+          match_branches,
+        );
+      let printed_branches = Doc.join(Doc.hardLine, items);
+
+      let printed_branches_after_brace =
+        Doc.concat([
+          force_break_if_line_comment(after_brace_comments, Doc.line),
+          printed_branches,
+        ]);
+
+      Doc.breakableGroup(
+        ~forceBreak=false,
```

Comment:
Pretty sure that a `breakableGroup(~forceBreak=false` is just a `group(`

Reply:
It is indeed :-D